reader.go

```diff
@@ -67,34 +68,35 @@ func (sfr *FactsetReader) download(filePath string, fileName string, dest string
 	return nil
 }
 
-func (sfr *FactsetReader) getLastVersion(files []os.FileInfo, searchedRes string) (string, error) {
-	recFile := &struct {
-		name string
-		vers int
+func (sfr *FactsetReader) getLastVersion(files []os.FileInfo, searchedFileName string) (string, error) {
+	foundFile := &struct {
+		name         string
+		majorVersion int
+		minorVersion int
 	}{}
 
-	r := regexp.MustCompile("[0-9]+\\.zip$")
 	for _, file := range files {
 		name := file.Name()
-		if !strings.Contains(name, searchedRes) {
+		if !strings.Contains(name, searchedFileName) {
 			continue
 		}
-		s := r.FindStringSubmatch(name)[0]
 
-		v, err := strconv.Atoi(strings.TrimSuffix(s, ".zip"))
+		fullVersion, err := GetFullVersion(name)
 		if err != nil {
-			return "", err
+			continue
 		}
 
-		if recFile.name == "" {
-			recFile.name = name
-			recFile.vers = v
-		} else if v > recFile.vers {
-			recFile.name = name
-			recFile.vers = v
+		majorVersion, _ := getMajorVersion(fullVersion)
+		minorVersion, _ := getMinorVersion(fullVersion)
+
+		if (majorVersion > foundFile.majorVersion) ||
+			(majorVersion == foundFile.majorVersion && minorVersion > foundFile.minorVersion) {
+			foundFile.name = name
+			foundFile.majorVersion = majorVersion
+			foundFile.minorVersion = minorVersion
 		}
 	}
-	return recFile.name, nil
+	return foundFile.name, nil
```

Comment:
There should be a check if the file was found at all, if it's a valid case. Also on the err!=nil to log something if it's handy, and not spammy.